internal/service/alarms/internal/alertmanager/alertmanager.go

```diff
@@ -61,3 +66,103 @@ func Setup(ctx context.Context, cl client.Client) error {
 	slog.Info("Successfully configured alertmanager")
 	return nil
 }
+
+// ConvertAmToAlarmEventRecordModels get alarmEventRecords based on the alertmanager notification and AlarmDefinition
+func ConvertAmToAlarmEventRecordModels(am *api.AlertmanagerNotification, aDefinitionRecords []models.AlarmDefinition) []models.AlarmEventRecord {
+	records := make([]models.AlarmEventRecord, 0, len(am.Alerts))
+	for _, alert := range am.Alerts {
+		slog.Info("Converting Alertmanager alert", "alert name", GetAlertName(*alert.Labels))
+		record := models.AlarmEventRecord{
+			AlarmRaisedTime:   *alert.StartsAt,
+			AlarmClearedTime:  alert.EndsAt,
+			PerceivedSeverity: getPerceivedSeverity(*alert.Labels),
+			AlarmStatus:       string(*alert.Status),
+			Fingerprint:       *alert.Fingerprint,
+			ResourceID:        GetResourceID(*alert.Labels),
+			ResourceTypeID:    GetResourceTypeID(*alert.Labels),
+		}
+
+		// Update Extensions with things we didn't really process
+		record.Extensions = getExtensions(*alert.Labels, *alert.Annotations)
+
+		// See if possible to pick up additional info from its definition
+		for _, def := range aDefinitionRecords {
+			if def.AlarmName == GetAlertName(*alert.Labels) { // TODO: match with the resourceTypeID too once init data is ready
```

Comment:
it will be possible to have the same alert name but from different resource type IDs (belong to different dicts). If 2 or more alerts with the same name and different managed cluster version come, it seems to me that they will be assigned the same alarmdefinitionID, am I right?

Reply:
yeah i think